Drop-In/src/androidTest/java/com/braintreepayments/api/CardDetailsFragmentUITest.kt

```diff
@@ -376,5 +376,6 @@ class CardDetailsFragmentUITest {
 
     fun whenStateIsRESUMED_whenThreeDSecureUserCancelationPresentInViewModel_showsSubmitButtonAgain() {
         // TODO: observe view model for UserCancelation cardTokenizationError and reset submit button
+        // TODO: is storing 3DS canceled in a view model necessary?
```

Comment:
We had a TODO for the approach of storing user canceled in view model but I think this can be done more simply by setting is loading to false in the view model from DropInActivity. Do you still think storing user canceled is necessary?

Reply:
This is almost UX territory. I feel like the current version finishes `DropInActivity` on user cancellation.

Reply:
I ran through the 3DS cancellation flow in the current version of Drop in and when a user cancels out of the 3DS screen they are returned to the card details screen (with the button clickable to submit again). That seems like the right flow to me, to make it easy for a user to re-try the same payment method, but do you think we should change that logic to Dismiss drop in if the user cancels?

Reply:
Ah yeah I remember now. We should definitely do the same thing yeah. I think since `DropInActivity` presents the `CardDetailsFragment`, it has knowledge that it exists. So what we can do is `findFragmentByTag` and call `CardDetailsFragment#notifyTokenizationError()` or something like that to notify the user and reset the button.